scripts/r.mask/r.mask.py

```diff
@@ -98,36 +98,52 @@ def main():
         gs.fatal(_("Either parameter <raster> or parameter <vector> is required"))
 
     mapset = gs.gisenv()["MAPSET"]
-    exists = bool(gs.find_file("MASK", element="cell", mapset=mapset)["file"])
+    mask_status = gs.parse_command("r.mask.status", format="json")
+    exists = mask_status["present"]
+    mask_full_name = mask_status["name"]
+
+    # Does the name point to the current mapset?
+    name, mask_mapset = mask_full_name.split("@", maxsplit=1)
+    if mask_mapset != mapset:
+        gs.fatal(
+            _(
+                "r.mask can only operate on raster mask in the current mapset "
+                "({current_mapset}), but mask name is set to {full_name}"
```

Comment:
why "but"?

Reply:
Full name will contain a different mapset than the current one, so the actual message should be more clear. Is that what you mean?

Reply:
ah! yes, fine, I was only reading the text, not the code before it :)